pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -96,9 +98,16 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 		}
 	}
 
-	err = validateIdentityUrl(doc.OpenShiftCluster, identityURL, isCreate)
-	if err != nil {
-		return nil, err
+	// Don't persist identity parameters in non-wimi clusters
+	if doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil || doc.OpenShiftCluster.Identity != nil {
+		if isCreate {
```

Comment:
Do we need to validate and persist identity URL in cases of update is well? My understanding is that we will still want to persist identity URL and tenant ID even in cases where it's not a new cluster creation, such as for an update, where we need to refresh client IDs for existing identities, or federate new identities. But I'm assuming that the secret URL/identity URL changes.

Reply:
If we don't have onboarding a new MSI at the time of update under scope, the identityURL and tenantID is not expected in that case. The credentials fetched via MSI Dataplane which will be persisted in KV will be the trusted source for the refreshing of the credentials.
The identityURL and TenantID will only be validated and persisted here(at frontend) in case of new MSI onboarding which is only scoped for the cluster creation I assume.

Reply:
Here's my understanding:

- We won't support updating the cluster MSI. Any attempt to remove, replace or assign additional should return an error to the user.
- Thus, we only need identity URL and tenant ID to fetch the credentials object and store in key vault during cluster creation.

This is in line with Rajdeep's points.

To address Caden's points:
- For any updates to the federated identities, we can just get the credential object from key vault and generate a token to perform federation to new identities.
- I'm not sure what you mean by "refresh client IDs" -> if a customer wants to update the federated identities, this isn't affected by the identity URL/tenant ID persistence. Any new identities we get can be federated by the cluster MSI, and we should have the backing credentials object in key vault.

Reply:
thanks for the clarification 👍